src/packages/cdk/src/app/database.ts

```diff
@@ -0,0 +1,36 @@
+import * as cdk from 'aws-cdk-lib';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { Construct } from 'constructs';
+
+import { GraphweaverAppConfig } from './types';
+
+export class DatabaseStack extends cdk.Stack {
+	public readonly dbInstance: rds.DatabaseInstance;
+
+	constructor(scope: Construct, id: string, config: GraphweaverAppConfig, props?: cdk.StackProps) {
+		super(scope, id, props);
+
+		// RDS PostgreSQL Instance
+		this.dbInstance = new rds.DatabaseInstance(this, `${id}-database`, {
```

Comment:
If they want to (optionally) pass us a database construct that matches the interface of things we actually need I'd be down for that as an option too.